### PR TITLE
PSG-2179 Test pipelines using minikube

### PR DESCRIPTION
**Pull request by jurajseffer**
- added CIrcleCI workflow that performs integration tests for Sars-Cov-2 and sythetic pipelines within minikube
- refactored minikube testing suite with new configs and updated scripts

### CODE REVIEW

File: minikube/run_pipelines.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+PSGA_ROOT_PATH="/app"
+
+for name in $PIPELINES; do
+  for config in $(ls test_configs/$name); do
+    echo "Running $name/$config test"
+    source test_configs/$name/$config
+    pipeline_pod="$( kubectl get pods -l app=$name-pipeline-minikube --no-headers -o custom-columns=':metadata.name' )"
+    kubectl -n psga-minikube exec $pipeline_pod -- bash -c "AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY} AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID} nextflow -log ${ANALYSIS_RUN}.log run ${PSGA_ROOT_PATH}/psga/main.nf --run ${ANALYSIS_RUN} --sequencing_technology ${SEQUENCING_TECHNOLOGY} --kit ${KIT} --metadata ${METADATA} --output_path ${OUTPUT_PATH}"
+    kubectl -n psga-minikube exec $pipeline_pod -- bash -c "cd ${PSGA_ROOT_PATH}/jenkins && pytest test_validation.py --expected-results-csv ${PSGA_ROOT_PATH}/jenkins/files/sars_cov_2/expected_results/${ANALYSIS_RUN}/results.csv --results-csv ${OUTPUT_PATH}/results.csv --output-path ${OUTPUT_PATH} --pathogen sars_cov_2 --sequencing-technology ${SEQUENCING_TECHNOLOGY} && cd -"
```
[the remark this answers is not in the document]
Comment (pdp10) [reply minikube/run_pipelines.sh:14]: Also note that we don't have integration tests for s-aureus here and I am working on the synthetic pipeline.

File: psga/common/common.config [hunk 2]
```diff
@@ -26,8 +26,8 @@ env {
     // process
     QUEUE_SIZE = "${QUEUE_SIZE}"
     PROCESS_MAX_RETRIES = "${PROCESS_MAX_RETRIES?:'3'}"
-    PROCESS_CPU_LOW = "${PROCESS_CPU_LOW?:'1'}"
-    PROCESS_CPU_HIGH = "${PROCESS_CPU_HIGH?:'2'}"
+    PROCESS_CPU_LOW = "${PROCESS_CPU_LOW?:''}"
+    PROCESS_CPU_HIGH = "${PROCESS_CPU_HIGH?:''}"
```
Comment (pdp10) [inline psga/common/common.config:30]: Why this change?
ncov-illumina requires 2 cpu to run.

Comment (pdp10) [reply psga/common/common.config:30]: Why did you set `PROCESS_CPU_LOW` to `''` by default?

File: minikube/test_configs/sars-cov-2/ont_unknown [hunk 3]
```diff
@@ -0,0 +1,6 @@
+METADATA="s3://synthetic-data-dev/UKHSA/pipeline_ci_tests/ont/metadata.csv"
```
Comment (pdp10) [inline minikube/test_configs/sars-cov-2/ont_unknown:1]: `s3://psga-test-static-data/pipeline_ci_tests/sars_cov_2/ont/unknown/metadata.csv`

File: minikube/test_configs/sars-cov-2/fasta [hunk 4]
```diff
@@ -0,0 +1,6 @@
+METADATA="s3://synthetic-data-dev/UKHSA/pipeline_ci_tests/unknown/metadata.csv"
```
Comment (pdp10) [inline minikube/test_configs/sars-cov-2/fasta:1]: `s3://psga-test-static-data/pipeline_ci_tests/synthetic/unknown/none/metadata.csv`

File: minikube/startup.sh [hunk 5]
```diff
@@ -22,9 +22,10 @@ kubectl config set-context minikube --namespace=psga-minikube
 echo "Setting service account"
 kubectl apply -f service_account.yaml
 
-kubectl create secret generic regcred \
-  --from-file=.dockerconfigjson=$HOME/.docker/config.json \
-  --type=kubernetes.io/dockerconfigjson
+# this raises an error if already present
+#kubectl create secret generic regcred \
+#  --from-file=.dockerconfigjson=$HOME/.docker/config.json \
+#  --type=kubernetes.io/dockerconfigjson
```
Comment (pdp10) [inline minikube/startup.sh:28]: I've commented this only because it raises an error if it is already present.
Is it possible to `apply` instead of `create`?

File: psga/common/common.config [hunk 6]
```diff
@@ -27,7 +27,7 @@ env {
     QUEUE_SIZE = "${QUEUE_SIZE}"
     PROCESS_MAX_RETRIES = "${PROCESS_MAX_RETRIES?:'3'}"
     PROCESS_CPU_LOW = "${PROCESS_CPU_LOW?:'1'}"
-    PROCESS_CPU_HIGH = "${PROCESS_CPU_HIGH?:'2'}"
+    PROCESS_CPU_HIGH = "${PROCESS_CPU_HIGH?:'1'}"
```
Comment (pdp10) [inline psga/common/common.config:30]: I think we should trigger a batch of 200 samples to determine the impact of this.

**Review by pdp10 — APPROVED** [review]
LGTM

File: psga/common/common.config (hunk 6 above)
Comment (jurajseffer) [reply psga/common/common.config]: Removed those changes, we can try them separately in isolation.
